extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py

```diff
@@ -0,0 +1,295 @@
+
+        event = CallbackEvent("on_serialize_complete", __name__, {"result": prompts})
+        await ai_config.callback_manager.run_callbacks(event)
+
+    async def deserialize(
+        self, prompt: Prompt, aiconfig: "AIConfigRuntime", params: Optional[Dict] = None
+    ) -> Dict:
+        """
+        Defines how to parse a prompt in the .aiconfig for a particular model
+        and constructs the completion params for that model.
+
+        Args:
+            serialized_data (str): Serialized data from the .aiconfig.
+
+        Returns:
+            dict: Model-specific completion parameters.
+        """
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_deserialize_start", __name__, {"prompt": prompt, "params": params}
+            )
+        )
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+
+        # Build Completion data
+        model_settings = self.get_model_settings(prompt, aiconfig)
+
+        completion_data = refine_chat_completion_params(model_settings)
+
+        messages = self._construct_chat_history(prompt, aiconfig, params)
+        messages.append({"role": "user", "parts": [{"text": resolved_prompt}]})
+
+        completion_data["contents"] = messages
+
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_deserialize_complete", __name__, {"output": completion_data}
+            )
+        )
+        return completion_data
+
+    async def run_inference(
+        self,
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        options: InferenceOptions,
+        parameters,
+    ) -> list[Output]:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            ExecuteResult: The response from the model.
+        """
+        await aiconfig.callback_manager.run_callbacks(
+            CallbackEvent(
+                "on_run_start",
+                __name__,
+                {"prompt": prompt, "options": options, "parameters": parameters},
+            )
+        )
+
+        # Auth check. don't need to explicitly set the key as long as this is set as an env var. genai.configure() will pick it up
+        get_api_key_from_environment("GOOGLE_API_KEY")
+        genai.configure()
+
+        # TODO: check and handle api key here
+        completion_data = await self.deserialize(prompt, aiconfig, parameters)
+
+        stream = True  # Default value
+        if options is not None and options.stream is not None:
+            stream = options.stream
+        elif "stream" in completion_data:
+            stream = completion_data["stream"]
+        completion_data["stream"] = stream
```

Comment:
Is this actually a supported param?

Reply:
Yep, there were 4 main attributes: 

<img width="503" alt="Screenshot 2023-12-14 at 5 26 46 PM" src="https://github.com/lastmile-ai/aiconfig/assets/141073967/0c3bf913-09ef-463c-a678-c96d88419257">